codexctl/device.py

```diff
@@ -305,75 +305,120 @@ def _read_version_from_path(self, ftp, base_path: str = "") -> tuple[str, bool]:
         os_release_path = f"{base_path}/etc/os-release" if base_path else "/etc/os-release"
 
         def file_exists(path: str) -> bool:
-            try:
-                ftp.stat(path)
-                return True
-            except FileNotFoundError:
-                return False
+            if ftp:
+                try:
+                    ftp.stat(path)
+                    return True
+                except FileNotFoundError:
+                    return False
+            return os.path.exists(path)
+
+        def read_file(path: str) -> str:
+            if ftp:
+                with ftp.file(path) as file:
+                    return file.read().decode("utf-8")
+            with open(path, encoding="utf-8") as file:
+                return file.read()

```

Comment:
```suggestion
        if ftp:
            def read_file(path: str) -> str:
                with ftp.file(path) as file:
                    return file.read().decode("utf-8")

        else:
            def read_file(path:str) -> str:
                with open(path, encoding="utf-8") as file:
                    return file.read()
```

Reply:
Better yet, merge the previous suggestion and this one to do the `if ftp` check once.